### PR TITLE
make get object testable

### PR DESCRIPTION
**Pull request by jackschofield23**
Signed-off-by: Jack Schofield <jack.schofield@answerdigital.com>

### CODE REVIEW

File: src/Storage/API/IStorageService.cs [hunk 1]
```diff
@@ -32,6 +32,15 @@ public interface IStorageService
         /// <returns>Task</returns>
         Task GetObjectAsync(string bucketName, string objectName, Action<Stream> callback, CancellationToken cancellationToken = default);
```
Comment (mocsharp) [inline src/Storage/API/IStorageService.cs:33]: We should remove this API to avoid confusion.

Comment (jackschofield23) [reply src/Storage/API/IStorageService.cs]: Removed

File: src/Plugins/AWSS3/AWS3StorageService.cs [hunk 2]
```diff
@@ -222,6 +220,20 @@ public async Task GetObjectWithCredentialsAsync(string bucketName, string object
             {
                 callback(obj.ResponseStream);
             }
+
+        }
+
+        public async Task<Stream> GetObjectWithCredentialsAsync(string bucketName, string objectName, Credentials credentials, CancellationToken cancellationToken = default)
```
Comment (mocsharp) [inline src/Plugins/AWSS3/AWS3StorageService.cs:226]: I think we should also remove one from line 210 GetObjectWithCredentialsAsync?

File: src/Storage/API/IStorageService.cs [hunk 3]
```diff
@@ -134,6 +133,17 @@ public interface IStorageService
...
+        /// <summary>
+        /// Downloads an objects as stream using temporary credentials.
+        /// </summary>
+        /// <param name="bucketName">Name of the bucket</param>
+        /// <param name="objectName">Name of the object in the bucket</param>
+        /// <param name="credentials">Temporary credentials used to connect</param>
+        /// <param name="callback">Action to be called when stream is ready</param>
+        /// <param name="cancellationToken">Optional cancellation token. Defaults to default(CancellationToken)</param>
+        /// <returns>Task</returns>
+        Task<Stream> GetObjectWithCredentialsAsync(string bucketName, string objectName, Credentials credentials, CancellationToken cancellationToken = default);
```
Comment (mocsharp) [inline src/Storage/API/IStorageService.cs:145]: May want to remove the other `GetObjectWithCredentialsAsync` method that takes a callback?

Comment (jackschofield23) [reply src/Storage/API/IStorageService.cs:134]: Removed

File: src/Plugins/AWSS3/AWS3StorageService.cs (hunk 2 above)
Comment (jackschofield23) [reply src/Plugins/AWSS3/AWS3StorageService.cs:210]: Removed

**Pull request merged**